go-controller/pkg/ovn/multicast.go

```diff
@@ -236,16 +252,150 @@ func (oc *DefaultNetworkController) createDefaultAllowMulticastPolicy() error {
 }
 
 // podDeleteAllowMulticastPolicy removes the pod's logical switch port from the
 // namespace's multicast port group. Caller must hold the namespace's
 // namespaceInfo object lock.
 func podDeleteAllowMulticastPolicy(nbClient libovsdbclient.Client, ns string, portUUID string) error {
-	return libovsdbops.DeletePortsFromPortGroup(nbClient, hashedPortGroup(ns), portUUID)
+	return libovsdbops.DeletePortsFromPortGroup(nbClient, getMulticastPortGroupName(ns), portUUID)
+}
+
+// updateStaleMulticastACLsDbIDs updates multicast ACLs that don't have new ExternalIDs set.
+// Must be run before WatchNamespace, since namespaceSync function uses syncNsMulticast, which relies on the new IDs.
+func (oc *DefaultNetworkController) updateStaleMulticastACLsDbIDs() error {
+	// Find stale multicast ACLs
+	legacyAclPred := libovsdbops.GetNoOwnerPredicate[*nbdb.ACL]()
+	legacyACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, legacyAclPred)
+	if err != nil {
+		return fmt.Errorf("unable to find multicast ACLs, cannot update stale data: %v", err)
+	}
+	if len(legacyACLs) == 0 {
+		return nil
+	}
+	mcastMatch := getMulticastACLMatch()
+	updatedACLs := []*nbdb.ACL{}
+	for _, acl := range legacyACLs {
+		var dbIDs *libovsdbops.DbObjectIDs
+		var aclT aclType
+		if acl.Match == mcastMatch {
+			// default deny
+			aclT = policyTypeStrToAclType(acl.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey])
+			dbIDs = getMcastACLDbIDs("", mcastDenyID, aclT, oc.controllerName)
+		} else if acl.Match == getACLMatch(types.ClusterRtrPortGroupName, mcastMatch, lportEgressAfterLB) {
+			// default allow egress
+			aclT = lportEgressAfterLB
+			dbIDs = getMcastACLDbIDs("", mcastAllowID, aclT, oc.controllerName)
+		} else if acl.Match == getACLMatch(types.ClusterRtrPortGroupName, mcastMatch, lportIngress) {
+			// default allow ingress
+			aclT = lportIngress
+			dbIDs = getMcastACLDbIDs("", mcastAllowID, aclT, oc.controllerName)
+		} else if strings.Contains(acl.Match, "(igmp || (ip4.src == ") ||
+			strings.Contains(acl.Match, "(mldv1 || mldv2 || (ip6.src == ") {
+			// ns allow ingress
+			aclT = lportIngress
+			// acl Name can be truncated (max length 64), but k8s namespace is limited to 63 symbols,
+			// therefore it is safe to extract it from the name
+			ns := strings.Split(libovsdbops.GetACLName(acl), "_")[0]
+			dbIDs = getMcastACLDbIDs(ns, mcastAllowID, aclT, oc.controllerName)
+		} else if strings.Contains(acl.Match, "ip4.mcast") ||
+			strings.Contains(acl.Match, "(mldv1 || mldv2 || "+ipv6DynamicMulticastMatch+")") {
+			// ns allow egress
+			aclT = lportEgressAfterLB
+			// acl Name can be truncated (max length 64), but k8s namespace is limited to 63 symbols,
+			// therefore it is safe to extract it from the name
+			ns := strings.Split(libovsdbops.GetACLName(acl), "_")[0]
+			dbIDs = getMcastACLDbIDs(ns, mcastAllowID, aclT, oc.controllerName)
+		} else {
+			continue
```

Comment:
so we are assigning owner IDs in these sync functions, every type of acl is synced in different function. This one only syncs multicast acls, so if we couldn't recognize the acl as multicast, we skip it hoping that other sync function, e.g. egress firewall sync will update it

Reply:
should we have a single "update acls" function that updates acls in a centralized place instead of splitting it over multiple functions throughout the code?